### PR TITLE
cli/tests: program_extend_program: Check for a specific error

### PR DESCRIPTION
**Pull request by illia-bobyr**
#### Problem
There are many errors that could be generated, creating a possibility for false positives.  We actually observed it in this test.

#### Summary of Changes
Specify the expected error.

### CODE REVIEW

File: cli/tests/program.rs [hunk 1]
```diff
@@ -1105,7 +1105,16 @@ fn test_cli_program_extend_program() {
         auto_extend: false,
         use_rpc: false,
     });
-    process_command(&config).unwrap_err();
+    let err = process_command(&config)
```
Comment (CriesofCarrots) [inline cli/tests/program.rs:1108]: Given what you found about this test, do we need to ensure we wait 1 slot (like on L1070) before processing this command? Or is the `account data too small` error always thrown before the already-modified-this-slot error?

**Review by CriesofCarrots — COMMENTED** [review]
Thanks, it will be great to check the specific error. Just one question

File: cli/tests/program.rs (hunk 1 above)
Comment (illia-bobyr) [reply cli/tests/program.rs:1108]: You have already modified it to `auto_extend: false`, in #1119, here:

https://github.com/anza-xyz/agave/commit/f255b31bdc4e3182c0e4e0706fdf5d0131d02a25#diff-cc498a8a54b2032afe95e65087e3603a45fe7855aea45fb0ef7f323ec4800f80R1105

So it is not going to try to extend the buffer anymore - which is correct.
It is a bug introduced in #791.

The race condition is only present when `auto_extend` is set to `true`.
It makes the command either fail with the "can not modify program in the same slot" error, or to succeed (actually less likely).
The test was written with an assumption that this command should fail with the "buffer is too small error".
But as the error check did not specify the exact error, and so any error caused the test to succeed.

After this PR, even if you enable `auto_extend`, the test will still correctly fail.
Either due to an unexpected error.  Or due to the command unexpectedly succeeding.

Comment (CriesofCarrots) [reply cli/tests/program.rs:1108]: Oh duh, okay. thanks for clarifying the timing.

**Review by CriesofCarrots — APPROVED** [review]

**Pull request merged**